packages/cli/src/config.test.ts

```diff
@@ -128,6 +128,35 @@ describe('getTypeScriptConfig', () => {
       'Failed to parse the TypeScript configuration.',
     );
   });
+
+  it.each([
+    {
+      module: 'Node16',
+      moduleResolution: 'NodeNext',
+    },
+    { module: 'NodeNext', moduleResolution: 'Node16' },
+    { module: 'CommonJS', moduleResolution: 'Node16' },
+    { module: 'ESNext', moduleResolution: 'NodeNext' },
+    { module: 'Node16' },
+    { moduleResolution: 'NodeNext' },
+  ])(
+    'throws an error if the `module` and `moduleResolution` options do not match',
+    (compilerOptions) => {
+      const { system } = getVirtualEnvironment({
+        files: {
+          '/index.ts': '// no-op',
+        },
+        tsconfig: {
+          compilerOptions,
+        },
+        checkDiagnostic: false,
```

Comment:
Setting `checkDiagnostic: false` since newer versions of TypeScript do this check as well, but we want to check it for older versions too.